tgui/packages/tgui/interfaces/Smes.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * @file
+ * @copyright 2020 Aleksej Komarov
+ * @author Original spookydonut (https://github.com/spookydonut)
+ * @author Changes Aleksej Komarov (https://github.com/stylemistake)
+ * @author Changes Sovexe (https://github.com/Sovexe)
+ * @license MIT
+ */
+
+import { useBackend } from '../backend';
+import { Box, Button, Flex, LabeledList, ProgressBar, Section, Slider } from '../components';
+import { formatPower } from '../format';
+import { Window } from '../layouts';
+
+// Common power multiplier
+const POWER_MUL = 1e3;
+
+export const Smes = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    capacityPercent,
+    charge,
+    formattedCharge,
+    inputAttempt,
+    inputting,
+    inputStartUp,
+    inputLevel,
+    inputLevelMax,
+    inputAvailable,
+    outputAttempt,
+    outputting,
+    outputLevel,
+    outputLevelMax,
+  } = data;
+  const inputState = (
+    capacityPercent >= 100 && 'good'
+    || (inputting && inputLevel) && 'average'
+    || 'bad'
+  );
+  const outputState = (
+    (outputAttempt && outputting) && 'good'
+    || charge > 0 && 'average'
+    || 'bad'
+  );
+  return (
+    <Window
+      width={340}
+      height={360}>
+      <Window.Content>
+        <Section title="Stored Energy">
+          <LabeledList>
+            <LabeledList.Item
+              label="Stored Power" >
+              {formattedCharge}J
+            </LabeledList.Item>
+          </LabeledList>
+          <ProgressBar
+            mt="0.5em"
+            value={capacityPercent * 0.01}
+            ranges={{
+              good: [0.5, Infinity],
+              average: [0.15, 0.5],
+              bad: [-Infinity, 0.15],
+            }} />
+        </Section>
+        <Section title="Input">
+          <LabeledList>
+            <LabeledList.Item
+              label="Charge Mode"
+              buttons={
+                <Button
+                  icon="power-off"
+                  color={inputAttempt ? "green" : "red"}
+                  onClick={() => act('tryinput')}>
+                  {inputAttempt ? 'On' : 'Off'}
+                </Button>
+              }>
+              <Box color={inputState}>
+                {capacityPercent >= 100 && 'Fully Charged'
+                  || (inputAttempt && inputLevel && !inputting) && 'Initializing'
+                  || (inputAttempt && inputLevel && inputting) && 'Charging'
+                  || (inputAttempt && inputting) && 'Idle'
+                  || 'Not Charging'}
```

Comment:
Nice way to avoid nested ternaries or having to declare a variable before the return and using here. 👍